PrestaShopCorp/ps_mbo
# Adds Guzzle cache and increase timeout

### Changes
- Upgrade phpstan (only used by GitHub Actions)
- Adds Guzzle cache to improve performance of `Module Selection` & `Theme Catalog` pages
- Increase timeout to fix issues encountered by merchants with slow network
https://www.prestashop.com/forums/topic/1030080-theme-catalog-error/
https://www.prestashop.com/forums/topic/1029064-prestashop-addon-impossible/
https://www.prestashop.com/forums/topic/1028687-have-trouble-connecting-to-addons/
https://www.prestashop.com/forums/topic/1027334-error-a-conectarse-a-addons-marketplace-en-prestashop-17/
https://www.prestashop.com/forums/topic/1025591-error-occur-when-get-recommend-add-on/
https://www.prestashop.com/forums/topic/1032084-catalogue-de-modules/

### Warning
**This module is available start PrestaShop 1.7.5.0 with PHP 5.6 minimum.**

So I cannot reuse PrestaShop core services introduced in 1.7.6.0 for Circuit Breaker and cache directory provider, that's why I create my owns.

--------------------------------------------
### How to test ?

Everything should works same as previously, `Module Selection` & `Theme Catalog` pages should be displayed as before.
If network is slow, we try to retrieve external content a little more longer before display "It looks like we have trouble connecting to Addons. Please refresh the page or check your firewall configuration."

#### Without network connection

Please note MBO cache is build on click on a recommended button or on display page with recommended modules after content or on display page with external content from Addons (Module Selection and Theme Catalog).

_If you clear Cache on Advanced Parameters > Performance without network connection with PrestaShop 1.7.6+, you should have an error but not related to MBO._

_I recommend to clear cache on Advanced Parameters > Performance with network connection then after cache clear confirmation you can disable network connection._

If MBO cache is not set and no network connection available or if we are unable to retrieve content before timeout :

- Migrated pages except Module Catalog page should display : `It looks like we have trouble connecting to Addons. Please refresh the page or check your firewall configuration.`
- Pages with a recommended button should display in modal : `Service Unavailable - It looks like we have trouble connecting to Addons. Please refresh the page or check your firewall configuration.`
- Pages with recommended modules displayed after content should display in modal : `Service Unavailable - It looks like we have trouble connecting to Addons. Please refresh the page or check your firewall configuration.`
